lib/private/assistant-client.ts

```diff
@@ -27,4 +29,40 @@ export class AssistantClientImp implements AssistantClient {
 	async search(req: SearchRequest): Promise<SearchResponse> {
 		return this.http.post('/search', { ...req, assistantId: this.id });
 	}
+	async *chat(req: ChatRequest): AsyncGenerator<ChatResponseType> {
+		const res = this.http.fetchAndStreamData('/chat', { ...req, assistantId: this.id });
+		if (req.stream) {
+			const decoder = new TextDecoder();
+			let current = '';
+
+			for await (const chunk of res) {
+				const uint8Array = new TextEncoder().encode(chunk);
+				const decodedChunk = decoder.decode(uint8Array, { stream: true });
+				const lines = decodedChunk.split('\n');
+				for (const line of lines.filter((l) => l.length > 0)) {
+					if (line.startsWith('data: ')) {
+						current = line.slice(6);
+						try {
+							const response: ChatResponseType = JSON.parse(current);
+							yield response;
+							current = '';
+						} catch (error) {
+							console.error('Error parsing response:', error);
+							throw new Error('Failed to parse response');
+						}
+					} else {
+						current += line;
+					}
+				}
+			}
+		} else {
+			let fullResponse = '';
+			for await (const chunk of res) {
```

Comment:
instead of doing this foreach, make another api method for chat answer without stream and just read all the response body at once